### PR TITLE
Changing implementation of FromSourceRegionZone

### PR DESCRIPTION
**Pull request by bathina2**
Reverts kanisterio/kanister#612

### CODE REVIEW

**Comment by bathina2** [comment]
Added UTs to test cross region migration as well

File: pkg/blockstorage/zone/zone.go [hunk 1]
```diff
@@ -79,12 +78,14 @@ func getAvailableZones(ctx context.Context, kubeCli kubernetes.Interface, validZ
 	if availableRegion != sourceRegion {
 		log.Info().Print("Source region and available region mismatch", field.M{"sourceRegion": sourceRegion, "availableRegion": availableRegion})
 	}
-	if len(availableZones) <= 0 { // Will never occur, NodeZonesAndRegion returns error if empty
-		log.Info().Print("No available zones found", field.M{"availableRegion": availableRegion})
+	// TODO: validate availableRegion
+	validZoneNames, err := m.FromRegion(ctx, availableRegion)
+	if err != nil || len(validZoneNames) == 0 {
```
Comment (tdmanv) [inline pkg/blockstorage/zone/zone.go]: what happens when `err != nil`?

Comment (bathina2) [reply pkg/blockstorage/zone/zone.go]: That is when we fail to get any zones from the available region (from k8s). We defer to using the source region.

Comment (tdmanv) [reply pkg/blockstorage/zone/zone.go]: Makes sense. Please add a comment since we're dropping the error.

**Pull request merged**